python/pyspark/sql/tests/test_functions.py

```diff
@@ -763,25 +798,55 @@ def test_higher_order_function_failures(self):
         from pyspark.sql.functions import col, transform
 
         # Should fail with varargs
-        with self.assertRaises(ValueError):
+        with self.assertRaises(PySparkException) as pe:
```

Comment:
should we use subclass PySparkValueError/PySparkTypeError in those tests?

Reply:
+1

Reply:
Sounds good.
Just updated the test, thanks!!